### PR TITLE
feat[wip]: migrate core to TypeScript

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

feat/refactor/

**Did you add tests for your changes?**
nno

**If relevant, did you update the documentation?**
no

**Summary**
- migrates core package to TS

**Does this PR introduce a breaking change?**
no

**Other information**

### CODE REVIEW

File: packages/webpack-cli/build/bin/cli.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    })
+        .then(() => {
+        logger.success(`${colors.bold('webpack')} was installed successfully.`);
+        bootstrap_1.default(process.argv, originalModuleCompile);
+    })
+        .catch(() => {
+        logger.error(`Action Interrupted, Please try once again or install ${colors.bold('webpack')} manually.`);
+        process.exit(2);
+    });
+}
```
Comment (alexander-akait) [inline packages/webpack-cli/build/bin/cli.js:35]: We should exclude these files

Comment (anshumanv) [reply packages/webpack-cli/build/bin/cli.js:35]: oh yep my mistake

**Comment by anshumanv** [comment]
will finish this tomorrow,

**Comment by anshumanv** [comment]
many conflicts, will reopen this late when it is priority

**Comment by alexander-akait** [comment]
@anshumanv we can start with typescipt jsdocs, it will be easy and faslty to merge step by step

**Comment by anshumanv** [comment]
yep lets go with ts jsdocs as first step

**Pull request closed without merging**